ddtrace/utils/formats.py

```diff
@@ -97,7 +98,25 @@ def parse_tags_str(tags_str):
     if not tags_str:
         return parsed_tags
 
-    for tag in tags_str.split(","):
+    # Identify separator based on which successfully identifies the correct
+    # number of valid tags
+    numtags = tags_str.count(":")
+    sep = (
+        ","
+        if sum(1 for t in tags_str.split(",") if ":" in t) == numtags
```

Comment:
can we write this in multi-line form, it took me _way_ too long to figure out what this statement was doing.

```python
sep = None
if sum(1 for t in tags_str.split(",") if ":" in t) == numtags:
    sep = ","
elif sum(1 for t in tags_str.split(" ") if ":" in t) == numtags:
    sep = " " 
```

I don't have a strong opinion here, but to me it reads a lot easier than trying to parse a double ternary.